OWML.ModHelper/ModConfig.cs

```diff
@@ -104,27 +108,82 @@ public void MakeConsistentWithDefaults(IModConfig defaultConfig)
                 return;
             }
 
-            AddMissingDefaults(defaultConfig);
+            var toRemove = Settings.Keys.Except(defaultConfig.Settings.Keys).ToList();
+            toRemove.ForEach(key => Settings.Remove(key));
 
-            var settingsCopy = new Dictionary<string, object>(Settings);
-            foreach (var setting in Settings)
+            var keysCopy = Settings.Keys.ToList();
+            foreach (var key in keysCopy)
             {
-                if (!defaultConfig.Settings.ContainsKey(setting.Key))
+                if (!IsSettingSameType(Settings[key], defaultConfig.Settings[key]))
                 {
-                    settingsCopy.Remove(setting.Key);
+                    TryUpdate(key, Settings[key], defaultConfig.Settings[key]);
                 }
-                else if (!IsSettingSameType(setting.Value, defaultConfig.Settings[setting.Key]))
+                else if (defaultConfig.Settings[key] is JObject objectValue && objectValue["type"].ToString() == "selector")
                 {
-                    settingsCopy[setting.Key] = defaultConfig.Settings[setting.Key];
+                    UpdateSelector(key, Settings[key], objectValue);
                 }
             }
-            Settings = settingsCopy;
+
+            AddMissingDefaults(defaultConfig);
+        }
+
+        private bool UpdateSelector(string key, object userSetting, JObject modderSetting)
```

Comment:
return value isn't used, and removing it would make it cleaner

Reply:
that's a setup for warning PR